src/console/MeasurementCollectionToFhir/Processor.cs

```diff
@@ -34,15 +38,50 @@ public class Processor : IEventConsumer
             _templateManager = templateManager;
             _measurementImportService = measurementImportService;
             _logger = logger;
+            _retryPolicy = CreateRetryPolicy(logger);
         }
 
         public async Task ConsumeAsync(IEnumerable<IEventMessage> events)
         {
             EnsureArg.IsNotNull(events);
             EnsureArg.IsNotNull(_templateDefinition);
 
-            var templateContent = _templateManager.GetTemplateAsString(_templateDefinition);
+            await _retryPolicy.ExecuteAsync(async () => await ConsumeAsyncImpl(events, _templateManager.GetTemplateAsString(_templateDefinition)));
+        }
+
+        private async Task ConsumeAsyncImpl(IEnumerable<IEventMessage> events, string templateContent)
+        {
             await _measurementImportService.ProcessEventsAsync(events, templateContent, _logger).ConfigureAwait(false);
         }
+
+        private static AsyncPolicy CreateRetryPolicy(ITelemetryLogger logger)
+        {
+            bool ExceptionRetryableFilter(Exception ee)
+            {
+                logger.LogError(new Exception("Encountered retryable exception", ee));
+                TrackExceptionMetric(ee, logger);
+                return true;
+            }
+
+            return Policy
+                .Handle<Exception>(ExceptionRetryableFilter)
+                .WaitAndRetryForeverAsync(retryCount => TimeSpan.FromSeconds(Math.Min(30, Math.Pow(2, retryCount))));
```

Comment:
> Math.Pow(2, retryCount) [](http://example.com/codeflow?start=89&length=24)

We will probably want to cap the max retry time.  Say 10 minutes?

Reply:
Just to confirm, you're saying that we no longer should retry forever, as we were doing before, and instead limit the overall retry operation to 10 minutes?

Reply:
No sorry, the intent is to cap the timespan for exponential back off.  I misread the code, I think we are covered.  Based on your above comment the longest we will wait for a retry is 30 seconds.